app/controllers/spree/api/v2/storefront/wished_products_controller.rb

```diff
@@ -0,0 +1,69 @@
+module Spree
+  module Api
+    module V2
+      module Storefront
+        class WishedProductsController < Spree::Api::V2::BaseController
+
+          helper Spree::Wishlists::ApiHelpers
+
+          def create
+            authorize! :create, Spree::WishedProduct
+
+            @wished_product = Spree::WishedProduct.new(wished_product_attributes)
+
+            current_wishlist_user = spree_current_user
+            @wishlist = current_wishlist_user.wishlists.find_by(id: @wished_product[:wishlist_id]) || current_wishlist_user.wishlist
+
+            if @wishlist.include? params[:wished_product][:variant_id]
+              @wished_product = @wishlist.wished_products.detect {|wp| wp.variant_id == params[:wished_product][:variant_id].to_i }
```

Comment:
@shradhaN what are we doing here?

Do we need to increase quantity in this case?

Reply:
This is not to increase the quantity, but to check if the particular product variant is already present in the wishlist. If so, then it will not add the product to the wishlist.